src/bosh_openstack_cpi/lib/cloud/openstack/cloud.rb

```diff
@@ -1029,5 +1029,23 @@ def validate_key_exists(keyname)
       @logger.debug("Using key-pair: `#{keypair.name}' (#{keypair.fingerprint})")
     end
 
+    def relative_and_validate_disk_size(flavor, resource_pool)
+      if resource_pool['root_disk'].nil?
+        if flavor.disk == 0
+          cloud_error("Flavor `#{resource_pool['instance_type']}' should have at least a 1GB boot disk")
```

Comment:
i would also in general try to avoid word "resource pool" in the cpi. it doesnt really exist at this level and is confusing when operators see it and they are using vm type in the director. "VM cloud properties" is more accurate.

Sent from my iPhone

> On May 13, 2016, at 7:22 AM, Marco Voelz <notifications@github.com> wrote:
> 
> In src/bosh_openstack_cpi/lib/cloud/openstack/cloud.rb:
> 
> > @@ -1029,5 +1029,23 @@ def validate_key_exists(keyname)
> >        @logger.debug("Using key-pair: `#{keypair.name}' (#{keypair.fingerprint})")
> >      end
> >  
> > +    def relative_and_validate_disk_size(flavor, resource_pool)
> > +      if resource_pool['root_disk'].nil?
> > +        if flavor.disk == 0
> > +          cloud_error("Flavor `#{resource_pool['instance_type']}' should have at least a 1GB boot disk")
> Some comments on the error msg:
> 
> don't use backticks (`), use single quotes instead (')
> change the msg to point to the new feature that is introduced here. How about "Flavor '#{resource_pool['instance_type']}' has a root disk size of 0. Either pick a different flavor or define root_disk.size in your resource_pool cloud_properties"
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
>